tools/azure-pipelines/jobs-template.yml

```diff
@@ -205,6 +225,19 @@ jobs:
       displayName: Cache E2E artifacts
       continueOnError: true
       condition: not(eq(variables['SKIP'], '1'))
+    - task: Cache@2
+      inputs:
+        key: 'e2e | $(TESTCONTAINER_CACHE_KEY)'
+        path: $(TESTCONTAINER_CACHE_FOLDER)
+        cacheHitVar: TESTCONTAINER_CACHE_HIT
+      continueOnError: true
+      condition: not(eq(variables['SKIP'], '1'))
+      displayName: Cache Testcontainer docker images
+    - script: |
+        docker load -i $(TESTCONTAINER_CACHE_FOLDER)/cache.tar || true
```

Comment:
nit: to deduplicate the name of the tar file this could be moved into a script. maybe we could unify this in `cache_docker_images.sh`, by having it accept `cache/load` as an argument to control what it should do.